lib/graphhopper/api_client.rb

```diff
@@ -277,7 +279,8 @@ def build_request_body(header_params, form_params, body)
       else
         data = nil
```

Comment:
What happens if we pass `nil` to `Zlib::Deflate.deflate`?

Reply:
```
tobi@speedy:~$ irb
irb(main):001:0> require 'zlib'
=> true
irb(main):002:0> Zlib::Deflate.deflate nil
Traceback (most recent call last):
        5: from /home/tobi/.asdf/installs/ruby/2.6.0/bin/irb:23:in `<main>'
        4: from /home/tobi/.asdf/installs/ruby/2.6.0/bin/irb:23:in `load'
        3: from /home/tobi/.asdf/installs/ruby/2.6.0/lib/ruby/gems/2.6.0/gems/irb-1.0.0/exe/irb:11:in `<top (required)>'
        2: from (irb):2
        1: from (irb):2:in `deflate'
TypeError (no implicit conversion of nil into String)
```

Good catch Adam.

Reply:
yeah, I didn't get that far, because gaphhopper keeps insisting that the request is not in GZIP format 😔